pkg/controller/resourcedistribution/resourcedistribution_controller.go

```diff
@@ -0,0 +1,391 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcedistribution
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"reflect"
+	"sync"
+
+	appsv1alpha1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util"
+	utildiscovery "github.com/openkruise/kruise/pkg/util/discovery"
+	"github.com/openkruise/kruise/pkg/util/ratelimiter"
+	utils "github.com/openkruise/kruise/pkg/webhook/resourcedistribution/validating"
+
+	"go.uber.org/atomic"
```

Comment:
Don't recommend to use this. Maybe just `sync/atomic` is enough.

Reply:
done